agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1806,6 +1806,33 @@ bool monitor_thread::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t event
     } break;
     case Event::STA_Connected: {
         LOG(TRACE) << "Received STA_Connected event";
+        auto msg = static_cast<bwl::sACTION_MONITOR_CLIENT_ASSOCIATED_NOTIFICATION *>(data);
+
+        auto sta_mac = tlvf::mac_to_string(msg->mac);
+        auto vap_id  = msg->vap_id;
+
+        LOG(INFO) << "STA_Connected: mac=" << sta_mac << " vap_id=" << vap_id;
+
+        mon_db.sta_erase(sta_mac);
```

Comment:
There's a bit a complicated race condition here...

The STAs IPv4 etc. will come from the controller via the "old flow". That old flow is triggered by the STA-CONNECTED event on the ap_manager. So, it is possible that the flow triggered by the ap_manager's STA-CONNECTED completes first (i.e. the STA was already added to the monitor through the VS message), and afterwards this function is triggered and removes the STA again. This way, we'll use the other information (IPv4 etc.) that is set by the VS flow.

Bottom line: I think we first have to do mon_db.find_sta to get the existing node, and copy all the info from it to the new node.

Or maybe it is easier to solve this issue by removing the STA-CONNECTED entirely from ap_manager and move it into monitor.

This can be done later, however, just create an issue for it.